src/core/cubic.c

```diff
@@ -397,21 +398,30 @@ CubicCongestionControlOnDataAcknowledged(
         // Slow Start
         //
 
-        //
-        // TODO: CongestionWindow should only be allowed to grow up to SlowStartThreshold
-        // here, and then any spare bytes should be handled in the Congestion Avoidance path.
-        //
-        Cubic->CongestionWindow += AckEvent->NumRetransmittableBytes;
+        Cubic->CongestionWindow += BytesAcked;
+        BytesAcked = 0;
         if (Cubic->CongestionWindow >= Cubic->SlowStartThreshold) {
             Cubic->TimeOfCongAvoidStart = CxPlatTimeMs64();
+
+            //
+            // We only want exponential growth up to SlowStartThreshold. If
+            // CongestionWindow has increased beyond SlowStartThreshold, set it back
+            // to SlowStartThreshold and treat the spare BytesAcked as if the bytes
+            // were acknowledged during Congestion Avoidance below.
```

Comment:
How would this work, considering `TimeInCongAvoid` would be `0`?

Reply:
The main point here is updating the AIMD window, which is dependent on the number of bytes ACKed instead of the amount of time in CA.

Reply:
Ok. Just merge my PR and main into this one. Assuming CI doesn't fail, we can merge it.